admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/ProductCustomPersistenceHandler.java

```diff
@@ -122,6 +125,9 @@ public Boolean canHandleRemove(PersistencePackage persistencePackage) {
 
     @Override
     public Boolean canHandleFetch(PersistencePackage persistencePackage) {
+        if (isRecursiveProductSelection(persistencePackage)) {
```

Comment:
maybe refactor to "return isRecursiveProductSelection(persistencePackage) || canHandleAdd(persistencePackage)"